### DISCUSSION TITLE
File history view filters

### DISCUSSION THREAD

**Discussion by VsevolodGolovanov** [gitextensions/gitextensions#9341] [Q&A]
I often use filters in the main repository window, namely by current branch and show first parents. Problem is, the filters state affects file history as well, where I don't want these filters like 99% of the time.
Is there a way to have a separate filters state for file history? Or to always open it with default filters?

**Comment by gerhardol** [comment]
Show first parents is a view setting and therefore follow.
Branch filter do not apply when opening FileHistory though.

It would be quite complex and confusing for most if settings were separate for the history, so I would not like to change that.

**Reply by VsevolodGolovanov** [reply]
I meant the "Show current branch only" setting, it applies too.

I get how different view settings could be confusing. But imho "Show first parents" specifically is confusing in File History in and of itself. You don't want it there, you want it to get an overview in the main repository window.
Like you said, the textual branch filter is treated differently - even though you can access it through `View -> Set advanced filter` both in the main repository window and in File History. Seems like it's a view setting as well. Yet it doesn't carry over.

In any case, wish it was an option to have two separate view settings states: for the main repository window and File History.

**Reply by VsevolodGolovanov** [reply]
Related: I often find it confusing when I double click a change in the Blame tab and it doesn't bring me to the relevant commit, because it's hidden by the view settings. It's a silent fail with no feedback. Wish it would reset Find History view filters automatically to show the commit. But not if the view settings are shared. Then it'd better somehow show a message explaining the fail.
